src/ArrayManager.cpp

```diff
@@ -163,9 +167,46 @@ void ArrayManager::resetTouch(void* pointer)
 
 
 void ArrayManager::resetTouch(PointerRecord* pointer_record) {
-  for (int i = 0; i < NUM_EXECUTION_SPACES; i++) {
-    pointer_record->m_touched[i] = false;
+  for (int space = CPU; space < NUM_EXECUTION_SPACES; ++space) {
+    pointer_record->m_touched[space] = false;
+  }
+}
+
+void ArrayManager::move(PointerRecord* record, ExecutionSpace space) 
+{
+  umpire::ResourceManager& rm = umpire::ResourceManager::getInstance();
+
+  if ( space == NONE ) {
+    return;
+  }
+
+#if defined(CHAI_ENABLE_UM)
```

Comment:
This is a hack, we should have a configurable ExecutionSpace<->MemorySpace mapping